### PR TITLE
A few small optimizations

### PR DESCRIPTION
**Pull request by cramforce**
- lazily init action map per element (most elements will never need it)
- clean up promise resolve for element loading state
- clean up elements from list of stubbed elements.

### CODE REVIEW

File: src/custom-element.js [hunk 1]
```diff
@@ -119,6 +119,8 @@ export function upgradeOrRegisterElement(win, name, toClass) {
     const element = stub.element;
     if (element.tagName.toLowerCase() == name) {
       tryUpgradeElementNoInline(element, toClass);
+      // Remove element from array.
+      stubbedElements.splice(i--, 1);
```
Comment (zhouyx) [inline src/custom-element.js]: What if `tryUpgradElementNoInline` fails, do also want to remove it from `stubbedElements`? 
Another thing is removing element and decrease the i here is not super straightforward, do you think that starting from the end of the array would be better?

Comment (cramforce) [inline src/custom-element.js]: Yeah, I actually think this is more correct. `upgrade` isn't idempotent, so calling it again doesn't make any sense. We don't really expect errors, so it should be fine either way.

**Comment by zhouyx** [comment]
LGTM

File: src/custom-element.js (hunk 1 above)
Comment (cramforce) [inline src/custom-element.js]: That is a good suggestion, but I'm pretty sure document-order upgrade is a good idea!

**Pull request merged**